### ISSUE TITLE
Type checking not working for object property value of the initializing object when setting the property from enum like [ EnumIds.id1 ]

### ISSUE THREAD

**Issue by fmflame** [Microsoft/TypeScript#18826]
**TypeScript Version:**  v2.5.3.

**Code**

```
enum Ids{
	id1= "id1",
	id2= "id2"
}

const IdValueMap: {[ id in Ids ]?: string} = {
	[ Ids.id1 ]: 1 // no type checking
};

IdValueMap[ Ids.id2 ] = 2; // type checking works and reports that the value 2 should be a string not number
```

**Expected behavior**
This should report an error that the value 2 should be a string not number:
```
const IdValueMap: {[ id in Ids ]?: string} = {
	[ Ids.id1 ]: 1
};
```

**Actual behavior**
No type checking.

**Comment by andy-ms** [comment]
Actually, I see a compiler error with your example.
```
src/a.ts(6,7): error TS2322: Type '{ [ Ids.id1 ]: number; }' is not assignable to type '{ id1?: string | undefined; id2?: string | undefined; }'.
  Types of property 'id1' are incompatible.
    Type 'number' is not assignable to type 'string | undefined'
```

**Comment by mhegazy** [comment]
Automatically closing this issue for housekeeping purposes. The issue labels indicate that it is unactionable at the moment or has already been addressed.
